packages/manager/modules/nasha/src/add/nasha-add.controller.js

```diff
@@ -55,12 +56,20 @@ export default class NashaAddCtrl {
     });
 
     this.offers = this.CucControllerHelper.request.getArrayLoader({
-      loaderFunction: () => this.NashaAddService.getOffers(),
+      loaderFunction: () =>
+        this.NashaAddService.getOffers().then((offers) => {
+          this.allOffers = offers;
+          return offers;
+        }),
     });
 
     this.durations = this.CucControllerHelper.request.getArrayLoader({
       loaderFunction: () => this.NashaAddService.getDurations(),
     });
+
+    this.catalog = this.CucControllerHelper.request.getArrayLoader({
+      loaderFunction: () => this.NashaAddService.getCatalog(),
+    });
```

Comment:
Can't we go without and use a promise instead ? Even though the rest is done this way

Reply:
You're right but should be handle as a global refacto in order to use new patterns.